### ISSUE TITLE
Fastcap ghost radius

### ISSUE THREAD

**Issue by ChillerDragon** [ddnet/ddnet#1172]
The ghost sometimes starts to replay in the FastCap mod even if the flag is not collected. The radius of the flag tile to trigger the ghost is probably too big.

**Comment by PathosEthosLogos** [comment]
@ChillerDragon Did you check whether the map has a start-line tile? DDNet recognises both the flag from CTF/FastCap and the start-line tile from Race.

**Comment by ChillerDragon** [comment]
Nope i recognized this on teerace maps and they dont use any startline stuff as far as i know.

**Comment by ChillerDragon** [comment]
I suspect the 32 from this line might be wrong.
https://github.com/ddnet/ddnet/blob/master/src/game/client/race.cpp#L77

**Comment by Robyt3** [comment]
Can't reproduce it on FastCap servers. I don't have the FastCap source code and it also seems that there are different versions of this mod currently being hosted. In vanilla the flag is grabbed when the distance between the tee and the flag is less than the sum of their physical sizes (14 + 28 = 42), so the 32 is technically wrong but in the opposite direction.

Permalink:

https://github.com/ddnet/ddnet/blob/50d0ecfda81838e66c03e1f7cd1d7f22101d6a84/src/game/client/race.cpp#L75

**Comment by ChillerDragon** [comment]
It is not easy to reproduce sadly. But that also makes it not that big of an issue.

**Comment by ChillerDragon** [comment]
A possible solution would be not detecting flag tile touches. But detect grabbing the flag (as in being the flag carrier). This should be the most portable. So it does not have to match the severs flag radius.

**Comment by ChillerDragon** [comment]
I tried to reproduce using ``move_raw`` and could not make it work. If I remember correctly I encountered that issue on [teerace](https://github.com/teerace/racemod/tree/5a4ed8c062deb855e5786cd6e3f499601cc200d5) servers back in the days. So I patched their mod to have the ddnet ``move_raw`` command and slid into the flag by spamming ``move_raw 1 0``. Could not reproduce it like that. Maybe it is some weird corner case (literally). Where one has to move through the corner of the flag. Or it is some miss prediction and requires network lag. Anyways I have not seen this bug in 5 years and the teerace servers are offline. I do not think it makes sense to implement my [proposed fix](https://github.com/ddnet/ddnet/issues/1172#issuecomment-1815301890) without being able to reproduce it. It might have other unwanted side effects such as the ghost being delayed until the first snap hits vs when the client predicts the flag touch.

I actually remember being super triggered about this bug because I always thought I managed to grab the flag but missed it by a pixel apparently. Anyways I will close this issue now until I encounter the bug again.
